internal/app/go-filecoin/plumbing/msg/waiter.go

```diff
@@ -100,23 +108,24 @@ func (w *Waiter) WaitPredicate(ctx context.Context, pred WaitPredicate, cb func(
 	return err
 }
 
-// Wait invokes the callback when a message with the given cid appears on chain.
-func (w *Waiter) Wait(ctx context.Context, msgCid cid.Cid, cb func(*block.Block, *types.SignedMessage, *vm.MessageReceipt) error) error {
+// Wait uses WaitPredicate to invoke the callback when a message with the given cid appears on chain.
+func (w *Waiter) Wait(ctx context.Context, msgCid cid.Cid, lookback uint64, cb func(*block.Block, *types.SignedMessage, *vm.MessageReceipt) error) error {
 	log.Infof("Calling Waiter.Wait CID: %s", msgCid.String())
 
 	pred := func(msg *types.SignedMessage, c cid.Cid) bool {
 		return c.Equals(msgCid)
 	}
 
-	return w.WaitPredicate(ctx, pred, cb)
+	return w.WaitPredicate(ctx, lookback, pred, cb)
 }
 
 // findMessage looks for a matching in the chain and returns the message,
 // block and receipt, when it is found. Returns the found message/block or nil
 // if now block with the given CID exists in the chain.
-func (w *Waiter) findMessage(ctx context.Context, ts block.TipSet, pred WaitPredicate) (*ChainMessage, bool, error) {
+// The lookback parameter is the number of tipsets this method will check before giving up.
```

Comment:
nit: the number of tipsets in the past this method will check before giving up

in my understanding the lookback shouldn't limit the total number of not yet synced blocks we wait for the message.